explorer/src/migration/1588016794171-SpeedUpJobRunSearch.ts

```diff
@@ -2,15 +2,25 @@ import { MigrationInterface, QueryRunner } from 'typeorm'
 
 export class SpeedUpJobRunSearch1588016794171 implements MigrationInterface {
   public async up(queryRunner: QueryRunner): Promise<any> {
-    await queryRunner.query(`
-        CREATE INDEX "job_run_searchable_addresses" ON "job_run" USING GIN
+    // NB: We have to do this in a different QueryRunner b/c the QueryRunner
+    // provided to us by the migration executor is already in a transaction.
+    // You can't run concurrent index creation inside a transaction.
+    // Note also that if this index creation fails, it will fail silently and
+    // simply be flagged INVALID and unused (see "Building Indexes Concurrently"
+    // here: https://www.postgresql.org/docs/9.1/sql-createindex.html)
+
+    const nonTransactionQueryRunner = queryRunner.connection.createQueryRunner()
+    await nonTransactionQueryRunner.connect()
+    await nonTransactionQueryRunner.query(`
+        CREATE INDEX CONCURRENTLY IF NOT EXISTS "job_run_searchable_addresses" ON "job_run" USING GIN
```

Comment:
Do we need `IF NOT EXISTS`?

Reply:
Feels like a good to have to me, since it's not happening in a transaction. Imagine if the index gets created but the migration fails for some other reason. The next time you start up you'll be blocked.